### PR TITLE
[border-routing] decide Router Solicitation state with `IsRouterSolicitationInProgress`

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: src/core/border_router/routing_manager.cpp [hunk 1]
```diff
@@ -636,7 +636,9 @@ void RoutingManager::StartRouterSolicitationDelay(void)
 {
     uint32_t randomDelay;
 
-    VerifyOrExit(!mRouterSolicitTimer.IsRunning() && mRouterSolicitCount == 0);
+    VerifyOrExit(!IsRouterSolicitationInProgress());
+
+    OT_ASSERT(mRouterSolicitCount == 0);
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:641]: @wgtdkp 
I think `mRouterSolicitCount == 0` is always true if `!mRouterSolicitTimer.IsRunning()` ?

Comment (wgtdkp) [inline src/core/border_router/routing_manager.cpp:641]:
```suggestion
```

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: src/core/border_router/routing_manager.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/core/border_router/routing_manager.cpp:641]: may still fine to keep the assert?

Comment (wgtdkp) [reply src/core/border_router/routing_manager.cpp:641]: `IsRouterSolicitationInProgress` is already making this for sure.
If we want to add the `assert`, better to add after every usage of `IsRouterSolicitationInProgress` but I suggest don't do that - it's meaningless :)

Comment (simonlingoogle) [reply src/core/border_router/routing_manager.cpp:641]: > `IsRouterSolicitationInProgress` is already making this for sure.

No, it isn't. 

`RouterSolicitationInProgress` means `mRouterSolicitTimer.IsRunning() || mRouterSolicitCount > 0`.

So, it's still possible that `mRouterSolicitCount > 0`. But here `mRouterSolicitCount` must be 0, so it's an additional condition.

**Review by jwhui — APPROVED** [review]

**Pull request merged**